### PR TITLE
Add default page selector

### PR DESCRIPTION
**Pull request by jacany**
Allows the user to select the default page to load (Posts, Convos, Global) on load.

Screenshot 2023-02-16 at 9 14 51 AM

### CODE REVIEW

File: packages/app/src/Pages/Root.tsx [hunk 1]
```diff
@@ -30,7 +30,16 @@ export default function RootPage() {
...
   };
-  const [tab, setTab] = useState<Tab>(RootTab.Posts);
+  const [tab, setTab] = useState<Tab>(() => {
+    switch (preferences.defaultPage) {
+      case "posts":
+        return RootTab.Posts;
+      case "conversations":
+        return RootTab.PostsAndReplies;
+      case "global":
+        return RootTab.Global;
```
Comment (fernandolguevara) [inline packages/app/src/Pages/Root.tsx:40]: add default case here

File: packages/app/src/State/Login.ts [hunk 2]
```diff
@@ -65,6 +65,11 @@ export interface UserPreferences {
    * Use imgproxy to optimize images
    */
   imgProxyConfig: ImgProxySettings | null;
+
+  /**
+   * Default page to select on load
+   */
+  defaultPage: "posts" | "conversations" | "global";
```
Comment (v0l) [inline packages/app/src/State/Login.ts:72]: `defaultRootTab`

File: packages/app/src/Pages/Root.tsx (hunk 1 above)
Comment (v0l) [reply packages/app/src/Pages/Root.tsx:40]: Yes just default to `Posts`

File: packages/app/src/Pages/Root.tsx [hunk 3]
```diff
@@ -30,7 +30,17 @@ export default function RootPage() {
...
-  const [tab, setTab] = useState<Tab>(RootTab.Posts);
+  const [tab, setTab] = useState<Tab>(() => {
+    switch (preferences.defaultRootTab) {
+      default:
+      case "posts":
+        return RootTab.Posts;
+      case "conversations":
+        return RootTab.PostsAndReplies;
+      case "global":
+        return RootTab.Global;
```
Comment (v0l) [inline packages/app/src/Pages/Root.tsx:41]:
```suggestion
      case "conversations":
        return RootTab.PostsAndReplies;
      case "global":
        return RootTab.Global;
      default:
        return RootTab.Posts;
```

**Review by v0l — APPROVED** [review]

**Pull request merged**